planner/core/rule_join_elimination.go

```diff
@@ -173,38 +183,74 @@ func (o *outerJoinEliminator) isDuplicateAgnosticAgg(p LogicalPlan) (_ bool, col
 			return false, nil
 		}
 		for _, expr := range aggDesc.Args {
-			if col, ok := expr.(*expression.Column); ok {
-				cols = append(cols, col)
-			} else {
-				return false, nil
-			}
+			// ExtractColumns will trans the expr to cols.
+			cols = append(cols, expression.ExtractColumns(expr)...)
 		}
 	}
 	return true, cols
 }
 
-func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Column, parentSchema *expression.Schema) (LogicalPlan, error) {
+func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Column, parentCols []*expression.Column) (LogicalPlan, error) {
+	var err error
+	var isEliminated bool
+	for join, isJoin := p.(*LogicalJoin); isJoin; join, isJoin = p.(*LogicalJoin) {
+		p, isEliminated, err = o.tryToEliminateOuterJoin(join, aggCols, parentCols)
+		if err != nil {
+			return p, err
+		}
+		if !isEliminated {
+			break
+		}
+		aggCols = aggCols[:0]
```

Comment:
Why clear it? What is the desired result if we eliminate outer join using empty agg cols in the next loop?

Reply:
After my serious consideration, it's better not to clear it. The `aggCols` means all columns in duplicate agnostic aggregate functions. They also shouldn't involve `groubByCols`. I'll change it.